### PR TITLE
Allow to pass custom prompt and provider configuration

### PR DESCRIPTION
**Pull request by christophwitzko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,22 +42,17 @@ const configuration = new Configuration({
 const openai = new OpenAIApi(configuration);
 
 const completion = await openai.createCompletion({
-  // the model will be ignored and the configured model from the project will be used
   model: "text-davinci-003",
-  // prompt: "Hello World!", // prompt is not needed anymore
-  // set this to you project id
+  prompt: "Hello World, my name is Alice!",
+  // you need to add your project id
```
Comment (heikir) [inline README.md]: Perhaps add why that's necessary? We will likely need some docs about this soonish.

File: client.ts [hunk 2]
```diff
@@ -79,14 +79,18 @@ export class Client {
...
+            ...this.options.defaultVariables,
+            ...config.variables,
+          }
+        : undefined,
       context: config.chatContext,
       userId: config.userId,
       truncateVariable:
         config.truncateVariable || this.options.defaultTruncateVariableConfig,
+      prompt: config.prompt,
+      providerConfig: config.providerConfig,
```
Comment (heikir) [inline client.ts:93]: Isn't providerConfig optional?

File: examples/openai.ts [hunk 3]
```diff
@@ -12,25 +12,18 @@ async function main() {
 
   const completion = await openai.createCompletion({
     model: "text-davinci-003",
-    // prompt: "Hello World!",
+    prompt: "Hello World, my name is Alice!",
```
Comment (heikir) [inline examples/openai.ts:15]: Perhaps a comment that you have to define it in case you are not using the CMS

File: openai/index.ts [hunk 4]
```diff
@@ -38,12 +37,28 @@ export class OpenAIApi extends originalOpenAI.OpenAIApi {
   ) {
     // return super.createChatCompletion(createChatCompletionRequest, options);
     const res = await this.cbClient.createCompletion({
-      variables: createChatCompletionRequest.variables || {},
       userId: createChatCompletionRequest.user,
       chatContext: {
         messages: createChatCompletionRequest.messages,
       },
       projectId: createChatCompletionRequest.projectId,
+      prompt: "\n", // we need to set prompt to avoid the default prompt
```
Comment (heikir) [inline openai/index.ts:45]: Again, perhaps "define prompt in case you manage your prompts in the code"

**Review by heikir — APPROVED** [review]
Looking good. Left some comments. Creating proper docs should be our prio in the next few weeks :)

File: client.ts (hunk 2 above)
Comment (christophwitzko) [reply client.ts:93]: yes, it can be undefined.

File: openai/index.ts (hunk 4 above)
Comment (christophwitzko) [reply openai/index.ts:45]: This is just a comment for us. If users are using the openai drop-in replacement we assume that they already manage their own prompts.

**Pull request merged**